### DISCUSSION TITLE
Pass errorWhileRendering to error page

### DISCUSSION THREAD

**Discussion by shahyar** [vikejs/vike#1077] [Discussion] [answered]
### Description

If you `guard` a page and `throw render(403, 'go away')` in it, the generic `_error/+Page.tsx` has no clue what the error was that brought it to this point. I can only know if  it `is404`.

From the examples, one could also `throw RenderErrorPage(....pageContext...)`, but that wouldn't allow a catch-all solution to error handling.

For now, I've added `'errorWhileRendering'` to `passToClient` and used the following code:

```typescript
const pageContext = usePageContext();
const err = pageContext.errorWhileRendering as any;

return <>{err._pageContextAbort._abortStatusCode} {err._pageContextAbort.abortReason}</>;
```

**Comment by brillout** [comment, marked as answer]
See docs about `abortReason` in https://vite-plugin-ssr.com/render.

**Reply by shahyar** [reply]
Thanks! Guess I should learn to read better. Is there anywhere to get the code though?

**Reply by brillout** [reply]
Done: https://github.com/brillout/vite-plugin-ssr/commit/1f7b92b9153f3f58f5d33e41678b10ace9e2386b (pre-released as `0.4.136-commit-1f7b92b`). You should now be able to access it as `pageContext.abortStatusCode`. (I didn't test it so let me know if it doesn't work.)

Btw. in case your company is up for it: https://github.com/sponsors/brillout.

**Reply by shahyar** [reply]
Thanks!

> Btw. in case your company is up for it

I'm still just investigating personally, but if we decide to use Vike, you can count on that.

**Reply by brillout** [reply]
Let know if you run into any issue or blocker.

Looking forward to further make sure things are working well for you. (FYI sponsors get a lot of high priority support.)
